docs/design/deviceshifu/design-deviceShifu-lwm2m.md

```diff
@@ -0,0 +1,211 @@
+# deviceShifu LwM2M Design
+
+deviceShifu LwM2M allow shifu connect device using [LwM2M protocol](https://omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/).
+
+This document outlines the design for integrating the [Lightweight Machine to Machine (LwM2M) protocol](https://omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/) into the deviceShifu framework to allow for the management of IoT devices.
+
+## Goals
+
+### Design Goal
+
+- Create a deviceshifu-lwm2m type to enable users to connect devices using the lwM2M protocol.
+- LwM2M protocol support both `read` and `write` requests.
+- Datagram Transport Layer Security (DTLS) support.
+- Resource observation or Notification.
+- [LwM2M protocol using v1.0.x version](https://www.openmobilealliance.org/release/LightweightM2M/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf).
+- LwM2M protocol under UDP.
+  
+### Design Non-Goal
+
+- Support LwM2M v1.1.x or later version.
+- Over TCP or other protocol.
+- Bootstrap Server.
+
+## Detail Design
+
+```mermaid
+flowchart LR
+subgraph kubernetes[Kubernetes]
+  subgraph ds[deviceShifu-LwM2M]
+          direction TB
+          lws[LwM2M Server]
+          http[HTTP Server]
+          http -->|call interface| lws
+  end
+  subgraph ds1[deviceShifu-LwM2M]
+          direction TB
+          lws1[LwM2M Server]
+          http1[HTTP Server]
+          http1 -->|call interface| lws1
+  end
+end
+ap[application]
+d[device]
+d1[device1]
+
+
+ap <-->|restful API| http
+ap <-->|restful API| http1
+lws <-->|lwM2M| d
+lws1 <-->|lwM2M| d1
+```
+
+The `deviceShifu-LwM2M` will represent each LwM2M object as an instruction that can be accessed through a RESTful API. The API will use the POST method for write operations and the GET method for read operations. The supported data formats for LwM2M v1.0 are TLV, JSON, Plain text, and Opaque.
+
+deviceShifu-LwM2M's LwM2M server will handle Register, Update, De-register, Read, Write, Observe and Notify.
+
+For each deviceshifu lwM2M should expose a UDP port for LwM2M communication.
+
+deviceShifu-LwM2M support two kind of mode normal and observe mode.
+- normal mode: just like the other deviceshifu, when call the instruction, deviceShifu will send Read or Write Request to deviceShifu and return response
+- observe mode: this mode will set the min and max notify time, then device will notify data when data changed or timeout. deviceShifu will record data, and return the data store in the memory cache when call the instruction. and this kind of mode also support read and write operation.
+
+deviceShifu will host a LwM2M server and listen on udp 3683(LwM2M default port) and http server(deviceshifu) on 8080.
+
+the LwM2M server will handle register, update, de-register request from device and maintain the device info in the memory cache.
```

Comment:
where is read/write/observe/notify request here? should be consist with above statement

Reply:
this line is same with above removed